src/emit.js

```diff
@@ -8,10 +8,27 @@ const defs: EventOptions = {
   composed: false
 };
 
+const hasNativeSupport = verifyNativeEventSupport();
+
+function verifyNativeEventSupport () {
+  try {
+    const detail = {};
+    const test = new CustomEvent('testEvent', { composed: true, detail });
+    return test.composed === true && test.detail === detail;
+  } catch (error) {
+    return false;
+  }
+}
+
 export function emit (elem: HTMLElement, name: string, opts: EventOptions): boolean {
   opts = { ...defs, ...opts };
-  const e: ComposedCustomEvent = document.createEvent('CustomEvent');
-  e.initCustomEvent(name, opts.bubbles, opts.cancelable, opts.detail);
-  Object.defineProperty(e, 'composed', { value: opts.composed });
+  let e: ComposedCustomEvent;
+  if (hasNativeSupport) {
+    e = new CustomEvent(name, opts);
+  } else {
+    e = document.createEvent('CustomEvent');
+    e.initCustomEvent(name, opts.bubbles, opts.cancelable, opts.detail);
+    Object.defineProperty(e, 'composed', { value: opts.composed });
```

Comment:
Technically we are already in polyfill-land with the original code. I believe it is still useful for browsers that cannot catch up. Since Edge and FF are likely to be implemented soon and IE11 is sliding fast, this may not be needed for too long. It's a shame, but since its still in heavy rotation here, we have to support IE11 for the time being. :angry: